app/models/answer_set.rb

```diff
@@ -102,6 +102,24 @@ def self.to_csv
     end
   end
 
+  def find_question_by_key_value(key, value, default=nil)
+    answer = self.answers.dig("questions")&.find {|h| h[key] == value}
+    return default if answer.nil?
+    answer.empty? ? default : answer
```

Comment:
Here we could use `answer.blank?` to handle the case in which `answer` is `nil`. This makes the previous line ```
return default if answer.nil?
```
superfluous. Same works for the other two methods below 😀